packages/reaction-product-variant/client/templates/products/productGrid/content/content.js

```diff
@@ -4,6 +4,8 @@
 
 Template.gridContent.helpers({
   displayPrice: function () {
-    return this._id && this.price.range;
+    if (this.price && this.price.range) {
```

Comment:
this.price can be null, undefined, a Number, or an Object because it's defined with two different schemas (product, productVariant).  In the grid though, it will always be an object, but it can be empty if the price was never set on the first variant (you'd only see this on a new product)